internal/cli/status.go

```diff
@@ -0,0 +1,845 @@
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithConfig(true), // optional config loading
+	); err != nil {
+		return 1
+	}
+
+	var ctxName string
+	defaultName, err := c.contextStorage.Default()
+	if err != nil {
+		c.ui.Output(
+			"Error getting default context: %s",
+			clierrors.Humanize(err),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+	ctxName = defaultName
+
+	ctxConfig, err := c.contextStorage.Load(ctxName)
+	if err != nil {
+		c.ui.Output("Error loading context %q: %s", ctxName, err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+	c.serverCtx = ctxConfig
+
+	cmdArgs := flagSet.Args()
+
+	if len(cmdArgs) > 1 {
+		c.ui.Output("No more than 1 argument required.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Determine which view to show based on user input
+	var projectTarget, appTarget string
+	if len(cmdArgs) >= 1 {
+		s := cmdArgs[0]
+		target := strings.Split(s, "/") // This breaks if we allow projects with "/" in the name
+
+		projectTarget = target[0]
+		if len(target) == 2 {
+			appTarget = target[1]
+		}
+
+	} else if len(cmdArgs) == 0 {
+		// If we're in a project dir, load the name. Otherwise we'll
+		// show a list of all projects and their status and leave projectTarget
+		// blank
+		if c.project.Ref() != nil {
+			projectTarget = c.project.Ref().Project
+		}
+	}
+
+	if appTarget == "" && c.flagApp != "" {
+		appTarget = c.flagApp
+	} else if appTarget != "" && c.flagApp != "" {
+		// setting app target and passing the flag app is a collision
+		c.ui.Output(wpAppFlagAndTargetIncludedMsg, terminal.WithWarningStyle())
+	}
+
+	// Generate a status view
+	if projectTarget == "" || c.flagAllProjects {
+		// Show high-level status of all projects
+		err = c.FormatProjectStatus()
+		if err != nil {
+			c.ui.Output("CLI failed to build project statuses:", terminal.WithErrorStyle())
+			c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
+	} else if projectTarget != "" && appTarget == "" {
+		// Show status of apps inside project
+		err = c.FormatProjectAppStatus(projectTarget)
+		if err != nil {
+			if status.Code(err) == codes.NotFound {
+				c.ui.Output(wpProjectNotFound, projectTarget, c.serverCtx.Server.Address, terminal.WithErrorStyle())
+			} else {
+				c.ui.Output("CLI failed to format project app statuses:", terminal.WithErrorStyle())
+				c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			}
+			return 1
+		}
+	} else if projectTarget != "" && appTarget != "" {
+		// Advanced view of a single app status
+		err = c.FormatAppStatus(projectTarget, appTarget)
+		if err != nil {
+			if status.Code(err) == codes.NotFound {
+				c.ui.Output(wpAppNotFound, appTarget, projectTarget, c.serverCtx.Server.Address, terminal.WithErrorStyle())
+			} else {
+				c.ui.Output("CLI failed to format app status", terminal.WithErrorStyle())
+				c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
```

Comment:
UX nit -- seeing both of these marked as an error makes me as a user think two errors happened, so I was confused for half a second. Can we combine them like `c.ui.Output("CLI failed to format app status: "+err.Error(), terminal.WithErrorStyle())`